RealmSwift/Sync.swift

```diff
@@ -905,7 +908,7 @@ public extension User {
     /// Refresh a user's custom data. This will, in effect, refresh the user's auth session.
     /// @returns A publisher that eventually return `Dictionary` with user's data or `Error`.
     func refreshCustomData() -> Future<[AnyHashable: Any], Error> {
```

Comment:
From my understanding a function should be marked as `@preconcurrency`, to tell Swift the this function was created without modern concurrency, so should' we make this one if we already marked the one using a completion block, maybe I'm misunderstanding this.

Reply:
`@preconcurrency` is only applicable to functions which take a `@Sendable` closure argument. Changing an argument from non-sendable to sendable is an API breaking change, which `@preconcurrency` fixes by making the function only require a sendable argument when in swift 6/strict concurrency mode, where the user has specifically opted into breaking changes.